packages/lodestar/src/network/gossip/validator.ts

```diff
@@ -45,10 +53,14 @@ export class GossipMessageValidator implements IGossipMessageValidator {
     if (await this.db.block.has(root)) {
       return false;
     }
-    return true;
-    //TODO: fix this
-    // const state = await this.db.state.getLatest();
-    // return verifyBlockSignature(this.config, state, signedBlock);
+    const headBlock = await this.db.block.getChainHead();
+    const state = await this.db.state.get(headBlock.message.stateRoot.valueOf() as Uint8Array);
+    const epoch = computeEpochAtSlot(this.config, signedBlock.message.slot);
+    const startSlot = computeStartSlotAtEpoch(this.config, epoch);
+    if(state.slot < startSlot) {
+      processSlots(this.config, state, startSlot);
+    }
```

Comment:
We are doing this in couple of places. It probably make sense to expose method in state db `getStateForSlot` which will fetch existing stored state for slot <= head.slot or process slots in case of slot > head.slot

Reply:
I think again, probably we don't need `processSlots` call above. Wdyt?

Reply:
I think you have to do it because proposers change on every epoch.